### PR TITLE
Migrate webapp to fastapi framework

### PR DESCRIPTION
**Pull request by linknum23**
- [x] Proof of concept migration (just get the app to work)
- [x] Migrate test framework to fastapi's
- [x] Convert API to use Pydantic models
  - [x] Zone class
  - [x] Group class
  - [x] Status class
  - [x] Preset class
  - [x] Source class
  - [x] Stream class hierarchy (only can be done partially until discriminator concept is fully supported, see https://github.com/samuelcolvin/pydantic/pull/2336 its close!)
- [x] Generate API Spec?
    - [x] Get basic swagger API output
    - [x] Migrate documentation from yaml to models.py
    - [x] Migrate tag documentation from yaml to _
    - [x] Add comments to add endpoint documentation
        - [x] zones
        - [x] groups
        - [x] sources
        - [x] streams
        - [x] presets
    - [x] Examples are not getting passed through yet!
    - [x] Add static examples
        - [x] zones
        - [x] groups
        - [x] sources
        - [x] streams
        - [x] presets
    - [x] Add static/generated example parameters
        - [x] zones
        - [x] groups
        - [x] sources
        - [x] streams
        - [x] presets
    - [x] Fix preset update nomenclature
    - [x] Serve the cool looking API documentation instead of the default one
    - [x] Serve openapi as human readable yaml
    - [x] save current documentation to a file so we can check it in to GH
    - [x] Add test to verify that each path has an example request data and response data

### CODE REVIEW

**Comment by linknum23** [comment]
@Lohrer and @micronova-jb Can you pull this and test the code to make sure everything works like you expect?

File: amplipi/app.py [hunk 1]
```diff
@@ -570,7 +574,8 @@ def specific_ctrl():
   return app
 
 if __name__ == "__main__":
-  parser = argparse.ArgumentParser(description='Create the openapi yaml file for AmpliPi')
+  """ Generate the openapi schema file in yamnl """
```
Comment (micronova-jb) [inline amplipi/app.py:577]: yamnl --> yaml

**Review by Lohrer — COMMENTED** [review]
Why the empty amplipi/py.typed?

The 'back' action results in interesting effects on the API. To reproduce go to (assuming localhost debug webserver) http://0.0.0.0:5000/api/doc which redirects to http://0.0.0.0:5000/api/doc#servers but it shows the Overview section (this is already weird). Then if I click on any link, say the first one 'Status -> Get Status' which leads to api/doc#get-/api and click back, it goes back to api/doc#servers. That was the last URL, so this makes sense, except now it actually shows the Servers section intead of Overview.

We're not done yet though: instead of the URL being api/doc#servers, it stays api/doc#get-/api! So then if you go forward, nothing happens - you're already at the api/doc#get-/api URL. Going back again also changes nothing now.

TLDR: URLs are getting overwritten and it's messing with my brain.

File: amplipi/app.py [hunk 2]
```diff
@@ -19,250 +19,564 @@
...
+      - [Stream](#tag--stream)
+      - [Preset](#tag--preset)
+
+    # OpenAPI
+
+    This API is documented using the OpenAPI specification
+"""
+
+@lru_cache(2)
+def create_yaml_doc(add_test_docs=True) -> str:
```
[the remark this answers is not in the document]
Comment (linknum23) [reply amplipi/app.py:483]: Plan is to use it to generate slightly modified docs for github

**Comment by linknum23** [comment]
@Lohrer I have added mypy to the github linting checks,. and started looking at a more verbose version of the output from pylint. All of the code changed in this PR is closer to what pylint expects. I don't know how much further I want to go with this though. Pylint generates a lot of messages...
